### PR TITLE
Add simple slides example

### PR DESCRIPTION
**Pull request by matklad**
Hi!

Yesterday I've spent some time trying to produce a PDF-presentation using ascidoctor-pdf.js, and the results look really great so far!

This is the template/CSS I came up with for producing minimal presentation. It might be useful to add a similar example to this repo, but keep in mind that I don't really know my way around CSS, so these might need some adjustment. 

The `asciidoctor.css` here is the same as in the `document` example, but with all media-queries removed. Given that PDF's produce a fixed-layout result, using media queries seems like a potential surprising footgun.

### CODE REVIEW

**Comment by Mogztter** [comment]
Thanks @matklad

This is pretty impressive, I love it :heart_eyes: 
I really like the page numbering with `${node.index + 1} / ${node.parent.blocks.length}` I didn't thought about it, that's clever!

> This is the template/CSS I came up with for producing minimal presentation. It might be useful to add a similar example to this repo, but keep in mind that I don't really know my way around CSS, so these might need some adjustment.

That's fine, we could improve the style a bit but overall this is really good.

> The asciidoctor.css here is the same as in the document example, but with all media-queries removed. Given that PDF's produce a fixed-layout result, using media queries seems like a potential surprising footgun.

:+1:

File: examples/slides/slides.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+= AsciiDocJs-PDF Presentation
```
Comment (Mogztter) [inline examples/slides/slides.adoc]: nit-pick, the capitalization is "Asciidoctor.js":

* AsciiDoc is the syntax 
* AsciiDoc Python is the deprecated Python implementation
* Asciidoctor is the Ruby implementation (default one now)
* Asciidoctor.js is the JavaScript implementation

File: examples/slides/template.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module.exports = {
```
Comment (Mogztter) [inline examples/slides/template.js]: Could you please add the command to generate the pdf in the README ?

File: examples/slides/template.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  paragraph: (node) => `<p class="${node.getRoles()}">${node.getContent()}</p>`,
+
+  section: (node) => `
+<section class="slide ${node.getRoles()} ${node.getTitle() == '!' ? 'no-title' : ''}">
+  <h2>${node.getTitle()}</h2>
+  <div class='slide-content'>
+    ${node.getContent()}
+  </div>
+  <footer>
+  <p>${node.index + 1} / ${node.parent.blocks.length}</p>
```
Comment (Mogztter) [inline examples/slides/template.js]: :+1:

File: examples/slides/slides.adoc (hunk 1 above)
Comment (matklad) [reply examples/slides/slides.adoc]: Note sure what's the proper way to fit PDF in there: choose `Asciidoctor.js PDF` to match https://asciidoctor.org/docs/asciidoctor-pdf/

File: examples/slides/template.js (hunk 2 above)
Comment (matklad) [reply examples/slides/template.js]: done

File: examples/slides/slides.adoc (hunk 1 above)
Comment (Mogztter) [reply examples/slides/slides.adoc]: "Asciidoctor.js PDF" is good.
But you could also use a short Rust presentation (if you want) and in this case use a "real" presentation title :smiley:

Comment (matklad) [reply examples/slides/slides.adoc]: >But you could also use a short Rust presentation (if you want) and in this case use a "real" presentation title smiley

Hehe, the reason why I am researching presentation tools is that I want to give a semester-long Rust course this spring :-)

Right now I am teaching Python, so hopefully I'll make a real presentation by Friday (using slides.com right now).

Comment (Mogztter) [reply examples/slides/slides.adoc]: > Hehe, the reason why I am researching presentation tools is that I want to give a semester-long Rust course this spring

I hope Asciidoctor.js PDF will work for you :wink: 

> Right now I am teaching Python, so hopefully I'll make a real presentation by Friday (using slides.com right now :( ).

Feel free to submit improvements to the slide template, I will gladly merge them :+1:

**Comment by matklad** [comment]
Looks gorgeous: https://github.com/matklad/pytesting-slides/blob/master/out/slides.pdf

**Comment by Mogztter** [comment]
Indeed :+1:

**Comment by matklad** [comment]
Ok, so here's the presentation I'll put in production in 20 minutes: https://github.com/matklad/pytesting-slides/blob/master/src/slides.pdf

asciidoctor-pdf.js makes me so productive, that I made 1.5x the amount of slides that can actually fit in the lecture :sweat_smile:

**Pull request merged**